anesthetic/samples.py

```diff
@@ -271,17 +272,19 @@ def plot_2d(self, axes, *args, **kwargs):
         else:
             fig = axes.values[~axes.isna()][0].figure
 
+        for pos in default_types:
+            if pos not in types:
+                types[pos] = None
+            if pos not in local_kwargs:
+                local_kwargs[pos] = {}
```

Comment:
If we are doing this change in handling types, would it make sense to do so instantly after `default_types` and adjusting all conditionals from `pos in types` to `types[pos] is not None`?

Reply:
That could be quite nice. You could also remove the rather clunky  filling in of the dictionary by using a [defaultdict](https://docs.python.org/2/library/collections.html#collections.defaultdict).

Reply:
not sure how to use defaultdict here